eventcatalog/src/components/MDX/ResourceLink/ResourceLink.astro

```diff
@@ -1,23 +1,50 @@
 ---
 import { buildUrl } from '@utils/url-builder';
-import { resourceToCollectionMap } from '@utils/collections/util';
+import { getItemsFromCollectionByIdAndSemverOrLatest, resourceToCollectionMap } from '@utils/collections/util';
+import { getCollection } from 'astro:content';
 
 const { id, version, type } = Astro.props;
 const collection = resourceToCollectionMap[type as keyof typeof resourceToCollectionMap];
+
+let slotHTML = await Astro.slots.render('default');
+let href = '#';
+let linkHasError = false;
+
+try {
+  if (collection === 'users' || collection === 'teams') {
+    const resources = await getCollection(collection, (resource) => resource.data.id === id);
+    const resource = resources[0];
+
+    slotHTML = slotHTML || resource.data.name;
+    href = buildUrl(`/docs/${collection}/${id}`);
+  } else {
+    const resourcesCollection = await getCollection(collection);
+
+    const resources = getItemsFromCollectionByIdAndSemverOrLatest(resourcesCollection, id, version);
+    const resource = resources[0];
+
+    slotHTML = slotHTML || resource.data.name;
+    href = buildUrl(`/docs/${collection}/${id}/${resource.data.version}`);
+  }
+} catch (error) {
+  console.error(`Failed to fetch related resource: ${id} of type ${type}`, error);
+  linkHasError = true;
+  slotHTML = id;
+}
 ---
 
 {
-  version && (
-    <a href={buildUrl(`/docs/${collection}/${id}/${version}`)} class="text-purple-500 hover:text-purple-700">
-      <slot />
+  linkHasError && (
+    <a href={href} class="text-red-500 hover:text-red-700">
+      ⚠️ {slotHTML} (broken link)
     </a>
   )
 }
 
 {
-  !version && (
-    <a href={buildUrl(`/docs/${collection}/${id}`)} class="text-purple-500 hover:text-purple-700">
-      <slot />
+  !linkHasError && (
+    <a href={href} class="text-purple-500 hover:text-purple-700">
+      {slotHTML}
```

Comment:
If the content inside <a> is on a separate line (like `<a>\n\t{content}\n</a>`), it ends up adding a trailing whitespace, which causes an unwanted underline at the end. There might be a better way to handle or prevent this.

Reply:
Ah OK, do you want me to hold off merging?

Reply:
Ignore the comment. There is no trailing underline this time 😮‍💨 
<img width="166" alt="Screenshot 2025-04-03 at 11 49 55" src="https://github.com/user-attachments/assets/906bd934-a847-4b45-8d1c-baca15815d8c" />

Reply:
Cool, shall we merge?

Reply:
Yes, please.
Honestly, I have no clue why the issue was happening in the first place 🤦

Reply:
Yeah no problem, thanks again for taking the time to improve for everyone!